x/auth/ante/sigverify.go

```diff
@@ -210,26 +210,33 @@ func (svd SigVerificationDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simul
 
 // authenticate the authentication of the TX for a specific tx signer.
 func (svd SigVerificationDecorator) authenticate(ctx sdk.Context, tx authsigning.Tx, signer []byte, sig signing.SignatureV2, txPubKey cryptotypes.PubKey) error {
-	acc, err := GetSignerAcc(ctx, svd.ak, signer)
-	if err != nil {
-		return err
+	// newlyCreated is a flag that indicates if the account was newly created
+	// this is only the case when the user is sending their first tx
+	newlyCreated := false
+	acc := GetSignerAcc(ctx, svd.ak, signer)
+	if acc == nil {
+		// if the account is nil, we assume this is the accounts first tx. in this case the account needs to be created, but
+		// the sign doc should keep account number 0. This is because the account number is not known until the account is created
+		// when the tx was signed, the account number was unknown and 0 was set.
```

Comment:
```suggestion
		// If the account is nil, we assume this is the account's first tx. In this case, the account needs to be
		// created, but the sign doc should use account number 0. This is because the account number is
		// not known until the account is created when the tx was signed, the account number was unknown
		// and 0 was set.
```